pkg/alertmanager/amcfg.go

```diff
@@ -888,6 +901,48 @@ func (cg *configGenerator) convertPushoverConfig(ctx context.Context, in monitor
 	return out, nil
 }
 
+func (cg *configGenerator) convertSnsConfig(ctx context.Context, in monitoringv1alpha1.SNSConfig, crKey types.NamespacedName) (*snsConfig, error) {
+	out := &snsConfig{
+		VSendResolved: in.SendResolved,
+		APIUrl:        in.ApiURL,
+		TopicARN:      in.TopicARN,
+		PhoneNumber:   in.PhoneNumber,
+		TargetARN:     in.TargetARN,
+		Subject:       in.Subject,
+		Message:       in.Message,
+		Attributes:    in.Attributes,
+	}
+
+	if in.HTTPConfig != nil {
+		httpConfig, err := cg.convertHTTPConfig(ctx, *in.HTTPConfig, crKey)
+		if err != nil {
+			return nil, err
+		}
+		out.HTTPConfig = httpConfig
+	}
+
+	if in.Sigv4 != nil {
+		accessKey, err := cg.store.GetSecretKey(ctx, crKey.Namespace, *in.Sigv4.AccessKey)
+		if err != nil {
+			return nil, errors.Errorf("failed to get secret %q", in.Sigv4.AccessKey)
```

Comment:
it would be better to wrap `err` since it should already contain the name and key of the reference

```suggestion
			return nil, errors.Wrap(err, "failed to get access key")
```